websites/D/Dropout/presence.ts

```diff
@@ -0,0 +1,199 @@
+import { QueryParams, IFrameData } from "./interfaces";
+
+const presence = new Presence({
+		clientId: "1240164154682249227",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://i.postimg.cc/kGkwmxKZ/dropout-logo.png",
+}
+
+let duration: number, currentTime: number, paused: boolean, playback: boolean;
+
+presence.on("iFrameData", (data: IFrameData) => {
+	playback = data.duration ? true : false;
+	if (playback) ({ duration, currentTime, paused } = data);
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname, href } = document.location,
+		showButtons = await presence.getSetting<boolean>("buttons"),
+		path = pathname.split("/");
+
+	path.shift();
+	if (pathname.endsWith("/")) path.pop();
+
+	getDetails(presenceData, path, showButtons, href);
+	presence.setActivity(presenceData);
+});
+
+function getDetails(
+	presenceData: PresenceData,
+	path: string[],
+	showButtons: boolean,
+	href: string
+): void {
+	switch (path[0].toLowerCase()) {
+		case "browse":
+			presenceData.details = "Browsing home page";
+			break;
+		case "continue-watching":
+			presenceData.details = "Browsing Continue Watching playlist";
+			break;
+		case "my-list":
+			presenceData.details = "Browisng my list";
+			break;
+		case "new-releases":
+			presenceData.details = "Browsing new releases";
+			break;
+		case "recommended-new-shelf":
+			presenceData.details = "Browsing trending page";
+			break;
+		case "series":
+			presenceData.details = "Browsing series";
+			break;
+		case "search":
+			getSearchDetails(presenceData);
+			break;
+		default:
+			getVideoDetails(presenceData, showButtons, href);
+	}
+}
+
+function getSearchDetails(presenceData: PresenceData): void {
+	presenceData.details = "Searching for:";
+	presenceData.state = parseQueryParams().q || "...";
+	presenceData.smallImageKey = Assets.Search;
```

Comment:
Since you are using modules, you will have to use a different name for your assets enum.

Reply:
Gotcha, thanks.